core/src/test/java/org/apache/iceberg/avro/TestAvroNameMapping.java

```diff
@@ -205,7 +205,7 @@ public void testMissingRequiredFields() {
     AssertHelpers.assertThrows(
         "Missing required field in nameMapping",
         IllegalArgumentException.class,
-        "Missing required field: x",
+        "Missing required field that doesn't have a default value: x",
```

Comment:
"with no default value"

Reply:
I think "Missing required field" is adequate. Based on where this error is placed, `Missing field with no default value` could be an ambiguous message.

Reply:
I'm just saying that it should be "with no default value" rather than saying "doesn't have a default value". We try to be as direct as possible.